Keas.Mvc/ClientApp/components/Keys/RevokeKeySerials.tsx

```diff
@@ -0,0 +1,130 @@
+        if (
+            nextProps.selectedKeySerial &&
+            (!this.props.selectedKeySerial ||
+                nextProps.selectedKeySerial.id !== this.props.selectedKeySerial.id)
+        ) {
+            this._fetchDetails(nextProps.selectedKeySerial.id);
+        }
+    }
+
+    public render() {
+        const { selectedKeySerial } = this.props;
+
+        if (!selectedKeySerial) {
+            return null;
+        }
+        debugger;
+        return (
+            <div>
+                <Modal
+                    isOpen={this.props.isModalOpen}
+                    toggle={this.props.closeModal}
+                    size="lg"
+                    className="keys-color"
+                >
+                    <div className="modal-header row justify-content-between">
+                        <h2>
+                            Revoke for {selectedKeySerial.key.code} {selectedKeySerial.number}
+                        </h2>
+                        <Button color="link" onClick={this.props.closeModal}>
+                            <i className="fas fa-times fa-lg" />
+                        </Button>
+                    </div>
+                    <ModalBody>
+                        <KeySerialEditValues
+                            keySerial={selectedKeySerial}
+                            disableEditing={true}
+                            openEditModal={this.props.openEditModal}
+                        />
+                        <KeySerialAssignmentValues
+                            selectedKeySerial={selectedKeySerial}
+                            openUpdateModal={this.props.openUpdateModal}
+                        />
+                    </ModalBody>
+                    <ModalFooter>
+                        <Button
+                            color="primary"
+                            onClick={() => this._revokeEquipment()}
+                            disabled={!this._isValidToRevoke() || this.state.submitting}
+                        >
+                            Revoke{" "}
+                            {this.state.submitting && <i className="fas fa-circle-notch fa-spin" />}
+                        </Button>{" "}
+                    </ModalFooter>
+                </Modal>
+            </div>
+        );
+    }
+
+    private _fetchDetails = async (id: number) => {
+        const url = `/api/${this.context.team.slug}/keySerials/details/${id}`;
+        let keySerial: IKeySerial = null;
+        try {
+            keySerial = await this.context.fetch(url);
+        } catch (err) {
+            toast.error("Error fetching key serial details. Please refresh to try again.");
+        }
+        this.props.updateSelectedKeySerial(keySerial);
+    };
+
+     private _revokeEquipment = async () => {
+        if (!this._isValidToRevoke()) {
+            return;
+        }
+        this.setState({ submitting: true });
+        await this.props.onRevoke(this.props.selectedKeySerial);
```

Comment:
i added this to the other revokes in my PR, but you might as well add try/catching in this one. 
```
this.setState({ submitting: true });
        try {
            await this.props.onRevoke(this.props.selectedKeySerial);
        } catch (err) {
            this.setState({ submitting: false });
            return;
        }
        this.setState({ submitting: false });
        this.props.closeModal();
```